pkg/resources/stage_grant.go

```diff
@@ -0,0 +1,153 @@
+package resources
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+)
+
+var ValidStagePrivileges = newPrivilegeSet(
+	privilegeUsage,
+)
+
+var stageGrantSchema = map[string]*schema.Schema{
+	"stage_name": &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "The name of the stage on which to grant privileges.",
+		ForceNew:    true,
+	},
+	"schema_name": &schema.Schema{
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     "PUBLIC",
```

Comment:
What is the reason for PUBLIC as a default? Defaulting to that level makes me a bit nervous that people will create stages that are more exposed than they expect. (I know that PUBLIC doesn't mean "public", but still).

Reply:
Yeah, I'll just remove the default. No real need for it.